Monads.NET/Maybe.cs

```diff
@@ -303,5 +369,26 @@ public static TInput Ignore<TInput>(this Tuple<TInput, Exception> input)
         {
             return input.Item1;
         }
+        
+
+        public static CastToSyntax<TInput> TryCast<TInput>(this TInput input)
+        {
+            return new CastToSyntax<TInput>(input);
+        }
+
+        public class CastToSyntax<TInput>
+        {
+            private readonly TInput _value;
+
+            internal CastToSyntax(TInput value)
+            {
+                _value = value;
+            }
+
+            public TOutput To<TOutput>() where TOutput : class,TInput
+            {
+                return _value as TOutput;
+            }
+        }
```

Comment:
The fuck is `CastToSyntax` doing here?

The `Try...` method convention in this library implies exception handling already.

Moreover, how is this simpler than `.With(i =>  i as IFucker)` ?